src/Runner.Common/JobServerQueue.cs

```diff
@@ -140,12 +142,12 @@ public void Start(Pipelines.AgentJobRequestMessage jobRequest)
             _fileUploadDequeueTask = ProcessFilesUploadQueueAsync();
 
             Trace.Info("Start results file upload queue.");
-            _summaryUploadDequeueTask = ProcessSummaryUploadQueueAsync();
+            _resultsUploadDequeueTask = ProcessResultsUploadQueueAsync();
```

Comment:
should we skip this if `_resultsClientInitiated == false`?

Reply:
If we don't enqueue then I think we can skip process the queue.  I can guard at enqueue too then we can skip this.

Reply:
hmmm, it looks pretty ugly to skip this as this is also called in `Shutdown`.

Do you think it's enough just to:
1. Skip enqueue any file
2. Make sure any file on the queue is dropped with no-op

If we have to remove processing this queue, the code looks pretty ugly. 😓

Reply:
Skip enqueue should be good enough.